components/match2/commons/starcraft_starcraft2/match_summary_starcraft_ffa.lua

```diff
@@ -9,470 +9,115 @@
-			:addClass('far fa-map')
-			:addClass('panel-content__game-schedule__icon')
-			:done()
-		:node(DisplayHelper.Map(game, {noLink = (game.map or ''):upper() == TBD}))
-end
-
----@param game table
----@return boolean
-function StarcraftMatchSummaryFfa._isFinished(game)
-	return game.winner ~= nil
 end
 
+---@param match table
 ---@param game table
----@param additionalClass string
----@return string?
-function StarcraftMatchSummaryFfa._countdownIcon(game, additionalClass)
-	local iconData = PHASE_ICONS[MatchGroupUtil.computeMatchPhase(game)] or {}
-	return Icon.makeIcon{iconName = iconData.iconName, color = iconData.color, additionalClasses = {additionalClass}}
-end
-
----Creates a countdown block for a given game
----Attaches any VODs of the game as well
----@param game table
----@return Html?
-function StarcraftMatchSummaryFfa._countdown(game)
-	if not game.extradata.timezoneid then
-		return
-	end
-
-	local timestamp = Date.readTimestamp(game.date) + (Timezone.getOffset(game.extradata.timezoneid) or 0)
-	local dateString = Date.formatTimestamp('F j, Y - H:i', timestamp) .. ' '
-			.. (Timezone.getTimezoneString(game.extradata.timezoneid) or UTC)
-
-	local stream = Table.merge(game.stream, {
-		date = dateString,
-		finished = StarcraftMatchSummaryFfa._isFinished(game) and 'true' or nil,
-	})
-
-	return mw.html.create('div'):addClass('match-countdown-block')
-			:node(require('Module:Countdown')._create(stream))
-			:node(game.vod and VodLink.display{vod = game.vod} or nil)
-end
-
----@param placementStart string|number|nil
----@param placementEnd string|number|nil
----@return string
-function StarcraftMatchSummaryFfa._displayRank(placementStart, placementEnd)
-	local places = {}
-
-	if placementStart then
-		table.insert(places, Ordinal.toOrdinal(placementStart))
-	end
-
-	if placementStart and placementEnd and placementEnd > placementStart then
-		table.insert(places, Ordinal.toOrdinal(placementEnd))
-	end
-
-	return table.concat(places, ' - ')
-end
-
----@param place integer
----@return string? icon
----@return string? iconColor
-function StarcraftMatchSummaryFfa._getTrophy(place)
-	if TROPHY_COLOR[place] then
-		return 'fas fa-trophy', TROPHY_COLOR[place]
-	end
-end
-
----@param opponent standardOpponent
----@return Html
-function StarcraftMatchSummaryFfa._displayOpponent(opponent)
-	return OpponentDisplay.BlockOpponent{
-		opponent = opponent,
-		showLink = true,
-		overflow = 'ellipsis',
-		teamStyle = 'hybrid',
+---@param gameIndex integer
+---@return table[]
+function Parser.gameHeader(match, game, gameIndex)
+	return {
+		HtmlWidgets.Div{
+			classes = {'panel-table__cell__game-title'},
+			children = {
+				MatchSummaryWidgets.CountdownIcon{game = game, additionalClasses = {'panel-table__cell-icon'}},
+				HtmlWidgets.Span{
+					classes = {'panel-table__cell-text'},
+					children = 'Game ' .. gameIndex
+				}
+			}
+		},
+		StarcraftMatchSummaryFfa._gamesHaveDates(match) and MatchSummaryWidgets.GameCountdown{game = game} or nil,
```

Comment:
Imo, add a check in the GameCountdown that it's a valid date instead

Reply:
but it usually is a valid date (it inherits it from match)
the checks checks if the date of all games and the match are the same or not

Reply:
fwiw renamed the function a bit